src/linter.h

```diff
@@ -50,60 +51,69 @@ class RuleVisitor : public NonRecursiveParseTreeVisitor {
   LinterResult GetResult() { return result_; }
 
  private:
-  std::function<LinterResult(const ASTNode *, const absl::string_view &)> rule_;
+  std::function<LinterResult(const ASTNode *, const absl::string_view &,
+                             const LinterOptions &)>
+      rule_;
   absl::string_view sql_;
   LinterResult result_;
+  LinterOptions option_;
 };
 
 // Stores properties of a single rule and menages possible
 // applications of this rule.
 class ASTNodeRule {
  public:
-  explicit ASTNodeRule(const std::function<
-                       LinterResult(const ASTNode *, const absl::string_view &)>
-                           rule)
+  explicit ASTNodeRule(
+      const std::function<LinterResult(
+          const ASTNode *, const absl::string_view &, const LinterOptions &)>
+          rule)
       : rule_(rule) {}
 
   // It applies the rule stored in this class to a sql statement.
-  LinterResult ApplyTo(absl::string_view sql);
+  LinterResult ApplyTo(absl::string_view sql, LinterOptions option);
 
  private:
-  std::function<LinterResult(const ASTNode *, const absl::string_view &)> rule_;
+  std::function<LinterResult(const ASTNode *, const absl::string_view &,
+                             const LinterOptions &)>
+      rule_;
 };
 
 // Debugger that will be erased later.
 LinterResult PrintASTTree(absl::string_view sql);
 
 // Checks if the number of characters in any line
 // exceed a certain treshold.
-LinterResult CheckLineLength(absl::string_view sql, int lineLimit = 100,
-                             const char delimeter = '\n');
+LinterResult CheckLineLength(absl::string_view sql,
+                             LinterOptions option = LinterOptions());
```

Comment:
In general, I don't think it is a good idea to make a default value for LinterOptions for each individual check. This way it is easy to make a mistake and not pass real LinterOptions to a check and suddenly it will be impossible to disable it.

Reply:
Yes, I will fix it.